corehq/apps/hqcase/api/updates.py

```diff
@@ -275,3 +300,82 @@ def _submit_case_blocks(case_blocks, domain, user, device_id):
         device_id=device_id,
         max_wait=15
     )
+
+
+def _get_owner_ids(domain, data, case_db):
+    owner_ids = []
+    case_ids = []
+    for data_item in data:
+        if data_item.is_new_case:
+            owner_ids.append(data_item.owner_id)
+        else:
+            case_id = data_item.get_case_id(case_db)
+            case_ids.append(case_id)
+
+        for name, index in data_item.indices.items():
+            index_case_id = index.get_id(case_db)
+
+            if index_case_id in case_db.real_case_ids:
+                case_ids.append(index_case_id)
+            else:
+                case_owner_id = case_db.get_owner_id(index_case_id)
+                owner_ids.append(case_owner_id)
+
+    case_owner_ids = (
+        CaseSearchES()
+        .domain(domain)
+        .case_ids(case_ids)
+        .values_list('owner_id', flat=True)
+    )
```

Comment:
Thinking through this part - what will happen if one or more of the case IDs passed in here is not found?

Reply:
This shouldn't be possible. The `case_ids` list is populated only if the conditional `index_case_id in case_db.real_case_ids` is met. If it isn't, then the ID simply gets appended to `owner_ids`. If an `owner_id` doesn't match either a user or a location, then this will get handled at the end of `validate_update_permission()` with an exception.